jvb/src/main/kotlin/org/jitsi/videobridge/cc/BandwidthProbing.kt

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.cc
+
+import org.jitsi.nlj.rtp.bandwidthestimation.BandwidthEstimator
+import org.jitsi.nlj.util.Bandwidth
+import org.jitsi.utils.concurrent.PeriodicRunnable
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging.TimeSeriesLogger
+import org.jitsi.videobridge.cc.allocation.BitrateControllerStatusSnapshot
+import org.jitsi.videobridge.cc.config.BandwidthProbingConfig
+import org.json.simple.JSONObject
+import java.util.function.Supplier
+import kotlin.random.Random
+
+class BandwidthProbing(
+    private val probingDataSender: ProbingDataSender,
+    private val statusSnapshotSupplier: Supplier<BitrateControllerStatusSnapshot>
+) : PeriodicRunnable(config.paddingPeriodMs), BandwidthEstimator.Listener {
+
+    /**
+     * The sequence number to use if probing with the JVB's SSRC.
+     */
+    private var seqNum = Random.nextInt(0xFFFF)
+
+    /**
+     * The RTP timestamp to use if probing with the JVB's SSRC.
+     */
+    private var ts: Long = Random.nextLong().and(0xFFFFFFFFL)
+
+    /**
+     * Whether or not probing is currently enabled
+     */
+    var enabled = false
+
+    /**
+     * The number of bytes left over from one run of probing to the next.  This
+     * avoids accumulated rounding errors causing us to under-shoot the probing
+     * bandwidth, and also handles the use when the number of bytes we want to
+     * send is less than the size of an RTP header.
+     */
+    private var bytesLeftOver = 0
+
+    private var latestBwe: Long = -1
+
+    var diagnosticsContext: DiagnosticContext? = null
+
+    override fun bandwidthEstimationChanged(newValue: Bandwidth) {
+        latestBwe = newValue.bps.toLong()
+    }
+
+    override fun run() {
+        super.run()
+        if (!enabled) {
+            return
+        }
+
+        // We calculate how much to probe for based on the total target bps
+        // (what we're able to reach), the total ideal bps (what we want to
+        // be able to reach) and the total current bps (what we currently send).
+        val bitrateControllerStatus = statusSnapshotSupplier.get()
+
+        // How much padding do we need?
+        val totalNeededBps = bitrateControllerStatus.currentIdealBps - bitrateControllerStatus.currentTargetBps
+        if (totalNeededBps < 1) {
+            // Don't need to send any probing
+            bytesLeftOver = 0
+            return
+        }
+
+        val latestBweCopy = latestBwe
+        if (bitrateControllerStatus.currentIdealBps <= latestBweCopy) {
+            // it seems like the ideal bps fits in the bandwidth estimation,
+            // let's update the bitrate controller.
+            // TODO(brian): this trigger for a bitratecontroller update seems awkward and may now be obsolete
+            //  since i now update it every time we get an updated estimate from bandwidth estimator
+//             dest.getBitrateController().update(bweBps);
```

Comment:
I guess update the commented-out code to Kotlin?

Reply:
It looks like this wasn't even valid in java anymore, as `dest` no longer exists.

Reply:
We can probably just remove this altogether?

Reply:
Yeah, that makes sense.  The place for code history is version control, not comments.